views/components/icons/index.jsx

```diff
@@ -0,0 +1,17 @@
+import { createElement, safe } from 'complate-stream'
+
+function ArrowRightDef () {
+  return <svg id="icon-arrow-right" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><defs><style>{`.cls-1{fill:none;stroke:currentColor;stroke-linecap:round;stroke-linejoin:round;stroke-width:1.5px;fill-rule:evenodd;}`}</style></defs><title>arrow-right-1</title><path class="cls-1" d="M5.5.75,16.22,11.47a.749.749,0,0,1,0,1.06L5.5,23.25"/></svg>;
+}
+
+export function IconDefinitions () {
+  return <svg class="icon-definitions" hidden>
+    <defs>
+      <ArrowRightDef />
```

Comment:
Just to be safe: This results in two nested `<svg>` elements - is that intended? It appears to work just fine (TIL!).

```html
<svg class="icon-definitions" hidden>
    <defs>
        <svg id="icon-arrow-right" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
            <defs>
                <style>.cls-1{fill:none;stroke:currentColor;stroke-linecap:round;stroke-linejoin:round;stroke-width:1.5px;fill-rule:evenodd;}</style>
            </defs>
            <title>arrow-right-1</title>
            <path class="cls-1" d="M5.5.75,16.22,11.47a.749.749,0,0,1,0,1.06L5.5,23.25"></path>
        </svg>
    </defs>
</svg>
```
```

Reply:
I implemented this using SVG Sprites which I read about in the book Inclusive Design Patterns. I did my first naive implementation (https://joyheron.com/responsive-nav/) using nested SVG elements, but I just went back to read the book, and he uses `<symbol>` instead of `<svg>`, so I will go back and look at that again.